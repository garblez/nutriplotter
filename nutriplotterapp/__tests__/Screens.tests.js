import 'react-native';
import React from 'react';
//import App from '../App';
import Home from '../screens/HomeScreen';
import Leaderboard from '../screens/LeaderboardScreen';
import Links from '../screens/LinksScreen';
import Login from '../screens/LoginScreen';
import SavedPlates from '../screens/SavedPlatesScreen';
import Score from '../screens/ScoreScreen';
import Settings from '../screens/SettingsScreen';

import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';


describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  //Screens for Home, Score and Login emitted due to use of variables set on other pages that can't be found without running first

  it('renders Leaderboard screen as expected', async () => {
    const tree = renderer.create(<Leaderboard />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders links screen as expected', async () => {
    const tree = renderer.create(<Links />).toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('renders SavedPlates screen as expected', async () => {
    const tree = renderer.create(<SavedPlates />).toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('renders settings screen as expected', async () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  })

});
