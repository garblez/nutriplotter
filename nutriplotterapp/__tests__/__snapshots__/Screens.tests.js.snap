// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders Leaderboard screen as expected 1`] = `
<View>
  <Text>
    waiting on leaderboards..
  </Text>
</View>
`;

exports[`App snapshot renders SavedPlates screen as expected 1`] = `null`;

exports[`App snapshot renders links screen as expected 1`] = `
<View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "padding": 15,
          "textAlign": "center",
        }
      }
    >
       
      The aim of the game is to build a nutritionally balanced meal. Use the search bar to find the foods you want to add.
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "padding": 15,
          "textAlign": "center",
        }
      }
    >
       
      A number of common foods are kept around the edge plate, these are: an apple, a slice of bread, an ice-cream and a can of soda.
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "padding": 15,
          "textAlign": "center",
        }
      }
    >
       
      Foods on your plate will be yellow in the search bar or coloured on the screen.
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Previous
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Tips for Balanced Meals
      </Text>
    </View>
  </View>
</View>
`;

exports[`App snapshot renders settings screen as expected 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 24,
            "textAlign": "center",
          }
        }
      >
        Delete Saved Plates
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 24,
            "textAlign": "center",
          }
        }
      >
        Delete Recently Searched Foods
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 4,
      }
    }
  />
</View>
`;
